### PR TITLE
Add the apps-external directory during setup only

### LINKED ISSUE #29839: [Feature Request] make the apps-external folder default used in new installations
**References:**
https://github.com/owncloud/core/issues/29807 (Move apps from apps to apps2 folder can cause issues)
https://github.com/owncloud/documentation/issues/3621 (Describe procedure how to move downloaded and installed apps from marketplace to the apps2 folder) 

**Background:**
When you have a new install, ``config.php`` gets created and it is important to have all the settings present respectively created early which may affect you when you go for upgrades later on. App management for apps not coming from core is one of those things.

When upgrading an installation via tar, it is a clean way to rename the old instance and extract into an empty folder. Copy your config.php and lucky when aliased your data directory. After doing all the steps to finalize, you just need to do an ./occ upgrade where core does the rest.
But what you have to do manually, identify all those apps which are in the old instance and copy them to the apps folder of the new instance and hopefully do not mix up things... This step is not necessary if the proposal below is implemented.

**Proposal:**
Adding as default into the created config.php the ability to use the ``apps-external`` folder.
see: https://doc.owncloud.org/server/latest/admin_manual/installation/apps_management_installation.html?highlight=apps2

```
  "apps_paths" => array (
      0 => array (
              "path"     => OC::$SERVERROOT."/apps",
              "url"      => "/apps",
              "writable" => false,
      ),
      1 => array (
              "path"     => OC::$SERVERROOT."/apps-external",
              "url"      => "/apps-external",
              "writable" => true,
      ),
  ),
````

**Benefit:**
You do not need to take care on moving/copying apps anymore.
You only need to either copy or alias the apps2 folder to the correct location.

@PVince81 @settermjd

### PR DESCRIPTION
**Pull request by mmattel**
## Description
- This PR adds the `apps-external` directory only during installation but not as before as part of the tar.
- Relates to changes made in master #30889 and stable10 #34656, both merged
Removing the changes made in Makefile,
Removing the physically created apps-external directory
Adopting the code in Setup.php

## Related Issue

- Fixes https://github.com/owncloud/core/issues/29839#issuecomment-476576489
([Feature Request] make the apps-external folder default used in new installations)

## Motivation and Context

Prevents re-adding that directory on every update.

## How Has This Been Tested?

- make test-php-unit TEST_DATABASE=mysql TEST_PHP_SUITE=tests/lib/SetupTest.php --> ok

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by PVince81** [comment]
- [x] TEST: no op when folder exists and no write permissions
- [ ] TEST: folder creates when not existing :no_entry_sign: FAIL
- [ ] TEST: error when no permission when attempting creation
- [ ] TEST: created folder has write permissions

**Comment by PVince81** [comment]
tested with occ maintenance:install ^

File: lib/private/Setup.php [hunk 1]
```diff
@@ -285,7 +285,22 @@ public function install($options) {
 		) {
 			$error[] = $l->t("Can't create or write into the data directory %s", [$dataDir]);
 		}
-		
+
+		// create the apps-external directory only during installation
+		$appsExternalDir = $this->config->getSystemValue('datadirectory', \OC::$SERVERROOT.'/apps-external');
```
Comment (PVince81) [inline lib/private/Setup.php]: key value is "datadirectory" here ??

File: lib/private/Setup.php [hunk 2]
```diff
@@ -285,7 +285,22 @@ public function install($options) {
...
+		if (!\file_exists($appsExternalDir)) {
+			@\mkdir($dataDir);
```
Comment (PVince81) [inline lib/private/Setup.php]: and here you create the data dir instead of the $appsExternalDir, seems all is a bit mixed up?

**Comment by mmattel** [comment]
```
sudo -uwww-data ./occ maintenance:install --database "mysql" --database-name "apptest" \
--database-host "localhost" --database-user "xyz" --database-pass "abc" \
--admin-user "admin" --admin-pass "def" 
ownCloud was successfully installed
```

**Comment by PVince81** [comment]
forgot to say that in my setup the setup goes through but the folder "apps-external" doesn't get created, because the `file_exists` check is done on the data folder, not on "apps-external" and returns true, so it doesn't attempt to create it.

**Comment by mmattel** [comment]
I fixed the issue, tested it, squashed and pushed, see my test from above.

**Comment by PVince81** [comment]
ah ok, didn't see the update. thanks

File: lib/private/Setup.php [hunk 3]
```diff
@@ -285,7 +285,22 @@ public function install($options) {
 		) {
 			$error[] = $l->t("Can't create or write into the data directory %s", [$dataDir]);
 		}
-		
+
+		// create the apps-external directory only during installation
+		$appsExternalDir = \OC::$SERVERROOT.'/apps-external';
+		if (!\file_exists($appsExternalDir)) {
```
Comment (PVince81) [inline lib/private/Setup.php]: seems we already do `is_dir` for existence check and `mkdir` in the block below, so this one here feel redundant ?

Comment (mmattel) [reply lib/private/Setup.php]: I just did the same as we do for the data directory to stay inline.
I can of course drop the if statement and just do @mkdir. Just le me know.

Comment (PVince81) [reply lib/private/Setup.php]: ah ok, fine then. it only happens once at setup time, so not a big issue

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
@mmattel please backport

**Comment by phil-davis** [comment]
Backport `stable10` #34902

**Pull request merged**